### PR TITLE
Added inline config options for #13

### PR DESCRIPTION
**Pull request by joezimjs**
When the update was made to pull the file to get the config, it removed
the ability to specify inline configuration. I re-added the inline config
possibility and added tests for inline configuration.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -11,8 +11,10 @@ module.exports = function (options, logger) {
 
 	delete options.failOnError;
 
-	if (options.config) {
+	if (typeof options.config === "string") {
```
Comment (SimenB) [inline index.js]: Could you do single-quotes for consistency?

File: index.js [hunk 2]
```diff
@@ -11,8 +11,10 @@ module.exports = function (options, logger) {
...
 		options = JSON.parse(fs.readFileSync(options.config));
+	} else if (typeof options.config === "object") {
+		options = options.config;
```
Comment (SimenB) [inline index.js]: Seems a bit odd to me that this should work, but the programmatic API of stylint is somewhat confusing, and your tests shows it working, so we're probably good :smile:

**Comment by SimenB** [comment]
Ah, now it makes more sense to me! You could probably remove the `delete options.failOnError;` and 

``` js
    if (Object.keys(options).length === 0) {
        options = undefined;
    }
```

as well then?

**Comment by joezimjs** [comment]
Those changes were made in my latest amendment that also got the tests fixed.

File: test.js [hunk 3]
```diff
@@ -79,6 +79,42 @@ it('It should log if file is invalid with custom options', function (cb) {
 	file('valid.styl');
 });
 
+it('It should not log if file is valid with custom inline options', function (cb) {
+	var log = sinon.spy();
+	stream = stylint({
+		config: {
+			"colons": "always",
```
Comment (SimenB) [inline test.js]: single quotes, please (and the keys doesn't have to be quoted).

My OCD kicks in on that...

**Comment by SimenB** [comment]
Awesome, this is great!

Could you update the README as well?

File: test.js (hunk 3 above)
Comment (joezimjs) [inline test.js]: Sorry about that. Just copied and pasted from the `config/.stylintrc file`.

Comment (SimenB) [inline test.js]: No problem! I just added the rule to JSHint, so in the future it'll break the build earlier :laughing:

**Comment by SimenB** [comment]
Looks great :+1: Squash up (and fix whitespace in readme), and I'll merge

File: readme.md [hunk 4]
```diff
@@ -60,7 +60,7 @@ Fail the gulp-process if a warning is issued
 
 ##### default logger function
 
-Type: `function`
```
Comment (SimenB) [inline readme.md:63]: Don't trim trailing space, it indicates linebreak in markdown.
Either re-add trailing space, or add an extra linefeed

**Comment by joezimjs** [comment]
There. That commit should finally do it :P

**Comment by SimenB** [comment]
Awesome, thanks a lot!

Published as `1.1.10`

**Pull request merged**